packages/venia-ui/lib/components/CartPage/ProductListing/productListing.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import gql from 'graphql-tag';
+import { useProductListing } from '@magento/peregrine/lib/talons/CartPage/ProductListing/useProductListing';
+
+import { mergeClasses } from '../../../classify';
+import LoadingIndicator from '../../LoadingIndicator';
+import defaultClasses from './productListing.css';
+import Product from './product';
+
+const ProductListing = props => {
+    const talonProps = useProductListing({ query: GET_PRODUCT_LISTING });
+    const { isLoading, items } = talonProps;
+
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    if (isLoading) {
+        return <LoadingIndicator>{`Fetching Cart...`}</LoadingIndicator>;
```

Comment:
Do we want a loading indicator here? I thought we were trying to get rid of them -- I just display nothing in summary when loading.

Reply:
It's really a case-by-case thing. You're right, we don't want loading indicators everywhere, but we also don't want to have a whole view suddenly just pop in. Ultimately, the way for us to avoid having spinners everywhere will have to be something like art direction but for loading states; until we have that, having an indicator here is fine.